### PR TITLE
Add comments to async query handlers

### PR DESCRIPTION
**Pull request by ykmr1224**
### Description
Added comments to async query handlers to make the responsibility of each handler clearer.
 
### Issues Resolved
N/A
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by seankao-az — DISMISSED** [review]

**Review by Swiddis — DISMISSED** [review]
Thanks! While you're at it, it might also be nice to [link to related classes](https://docs.oracle.com/javase/1.5.0/docs/tooldocs/windows/javadoc.html#@see), to make it slightly easier to navigate if someone e.g. lands on `IndexDMLResultStorageService` without having seen `IndexDMLResultStorage`.

**Comment by ykmr1224** [comment]
> Thanks! While you're at it, it might also be nice to [link to related classes](https://docs.oracle.com/javase/1.5.0/docs/tooldocs/windows/javadoc.html#@see), to make it slightly easier to navigate if someone e.g. lands on `IndexDMLResultStorageService` without having seen `IndexDMLResultStorage`.

That's a good point. I'm wondering if we need those links in the comments considering IDE or Github provide code navigation feature. I think the link would be beneficial when we want to mention some related class which is not obviously referred in the code.

File: spark/src/main/java/org/opensearch/sql/spark/dispatcher/BatchQueryHandler.java [hunk 1]
```diff
@@ -28,6 +28,10 @@
 import org.opensearch.sql.spark.leasemanager.model.LeaseRequest;
 import org.opensearch.sql.spark.response.JobExecutionResponseReader;
 
+/**
+ * The handler for batch query. With batch query, queries are executed as single batch. The queries
+ * are sent along with job execution request to spark.
```
Comment (noCharger) [inline spark/src/main/java/org/opensearch/sql/spark/dispatcher/BatchQueryHandler.java:33]: what is job execution request? can you add a ref link to the class?

File: spark/src/main/java/org/opensearch/sql/spark/dispatcher/IndexDMLHandler.java [hunk 2]
```diff
@@ -31,7 +31,12 @@
 import org.opensearch.sql.spark.flint.operation.FlintIndexOpFactory;
 import org.opensearch.sql.spark.response.JobExecutionResponseReader;
 
-/** Handle Index DML query. includes * DROP * ALT? */
+/**
+ * The handler for Index DML (Data Manipulation Language) query. Handles DROP/ALT/VACUUM operation
```
Comment (noCharger) [inline spark/src/main/java/org/opensearch/sql/spark/dispatcher/IndexDMLHandler.java:35]: ALT -> ALTER?

File: spark/src/main/java/org/opensearch/sql/spark/dispatcher/BatchQueryHandler.java (hunk 1 above)
Comment (ykmr1224) [reply spark/src/main/java/org/opensearch/sql/spark/dispatcher/BatchQueryHandler.java]: Fixed.

File: spark/src/main/java/org/opensearch/sql/spark/dispatcher/IndexDMLHandler.java (hunk 2 above)
Comment (ykmr1224) [reply spark/src/main/java/org/opensearch/sql/spark/dispatcher/IndexDMLHandler.java]: Good catch. Fixed.

**Review by noCharger — APPROVED** [review]
Thanks for the change!

**Review by seankao-az — APPROVED** [review]

**Review by Swiddis — APPROVED** [review]

**Pull request merged**